Dev

Bivis63 opened the pull request.

Bivis63 committed:
дз спринт 22 (задание 1. добавил новый фрагмент , сделал верстку экрана и логику поведения экрана)

Bivis63 committed:
дз спринт 22 (задание 2 выполнил , приступил к 3 заданию)

Bivis63 committed:
дз спринт 22 (задиние 3 делаю навигацию)

Bivis63 committed:
дз спринт 22 (задиние 3 делаю навигацию,1.2)

Bivis63 committed:
дз спринт 22 (задиние 3 начал , начал делать добавление трека в плейлист)

Bivis63 committed:
дз спринт 22 (реализованно добавление трека в плей листы)

DmitrijEgorow commented on .idea/deploymentTargetDropDown.xml:
Стоит настроить gitignore для .idea

DmitrijEgorow commented on app/src/main/AndroidManifest.xml:
```diff
@@ -4,6 +4,9 @@
+    <uses-permission android:name="android.permission.CAMERA"/>
 
     <application
         android:name=".util.App"
```
Странно, что App в подпакете util

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/main/MainActivity.kt:
```diff
@@ -3,26 +3,40 @@ package com.example.playlistmaker.main
 
     lateinit var binding: ActivityMainBinding
 
+    lateinit var navigationController: NavController
```
Зачем эти поля видны извне класса?

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/media/ui/NewPlayLists/NewPlayListFragment.kt:
```diff
@@ -0,0 +1,170 @@
+        )
+    }
+
+    private fun saveImageToPrivateStorage(uri: Uri) {
```
Это работа с файлами (data), её нужно перенести из ui

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/media/ui/NewPlayLists/NewPlayListFragment.kt:
```diff
@@ -0,0 +1,170 @@
+    private fun openDialog() {
+        MaterialAlertDialogBuilder(requireContext())
+            .setTitle("Завершить создание плейлиста?")
+            .setMessage("Все несохраненные данные будут потеряны")
```
Лучше вынести в строковые ресурсы

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/media/ui/NewPlayLists/NewPlayListFragment.kt:
```diff
@@ -0,0 +1,170 @@
+            addToDB(selectedImageUri)
+            Toast.makeText(
+                requireContext(),
+                "Плейлист ${binding.textInputEditText.text} создан",
```
Лучше вынести в строковые ресурсы

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/media/ui/NewPlayLists/NewPlayListFragment.kt:
```diff
@@ -0,0 +1,170 @@
+    private lateinit var binding: FragmentNewPlayListBinding
+    private var showDialog: Boolean = false
+    private val viewModel by viewModel<NewPlayListViewModel>()
+    var selectedImageUri: Uri? = null
```
Это поле тоже доступно извне класса

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/media/ui/NewPlayLists/NewPlayListFragment.kt:
```diff
@@ -0,0 +1,170 @@
+        }
+    }
+
+    private fun addToDB(imageUri: Uri?) {
```
Фрагмент не должен знать, куда именно сохраняются данные (в бд / в файл / ...)

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/player/ui/fragment/AudioPlayerFragment.kt:
```diff
@@ -0,0 +1,246 @@
+        val onBackPressedDispatcher = requireActivity().onBackPressedDispatcher
+        val item = arguments?.getSerializable(ITEM) as Track
+
+        track =item
```
Форматирование стоит поправить

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/player/ui/fragment/AudioPlayerFragment.kt:
```diff
@@ -0,0 +1,246 @@
+        if (viewModel.clickDebounce()) {
+            if (!viewModel.isInPlaylist(playlist = playlistModel, trackId = track.trackId.toLong())) {
+                viewModel.addToPlaylist(playlist = playlistModel, track = track)
+                Toast.makeText(requireContext().applicationContext, "Трек добавлен в ${playlistModel.name}", Toast.LENGTH_SHORT)
```
Лучше вынести в строковые ресурсы

DmitrijEgorow commented on app/src/main/res/layout/fragment_audio_player.xml:
```diff
@@ -0,0 +1,272 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+
+<androidx.coordinatorlayout.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Стоит убрать пустые строки

DmitrijEgorow commented on app/src/main/res/layout/fragment_play_lists.xml:
Нижние элементы списка видны не полностью


https://github.com/Bivis63/Playlist_Maker/assets/36837067/bd2f6d6b-4ad7-4795-a089-ffa10efaf908

DmitrijEgorow commented on app/src/main/res/layout/fragment_track_search.xml:
На устройствах с увеличенным размером шрифта выглядит так:

![Screenshot_20230909_181914](https://github.com/Bivis63/Playlist_Maker/assets/36837067/c3721c60-66bc-48da-a2e2-7bcb0ce04207)

DmitrijEgorow commented:
Скопированный файл с изображением для обложки не используется в приложении


https://github.com/Bivis63/Playlist_Maker/assets/36837067/ff27774c-64d8-4ed3-86a8-3fd98e022757

Bivis63 committed:
дз спринт 22 (исправил логику работы bottom sheet при добавление трека,
иправил отображение плейлистов,добавил обработку счетчика треков, вынес логику сохранения картинки из ui слоя )

Bivis63 committed:
дз спринт 22 (Файл с изображением для обложки плейлиста при создании копируется в хранилище, доступное только приложению Playlist Maker.)

Bivis63 committed:
дз спринт 22 (добавил при нажатии на кнопку «Новый плейлист» кно добавления трека в плейлист исчезает, и если текущий трек добавлен в выбранный плейлист, то окно добавления трека в плейлист не исчезает )
